src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -380,21 +381,24 @@ export class MockHttpSocket extends MockSocket {
 
     if (response.body) {
       try {
-        const reader = response.body.getReader()
-
-        while (true) {
-          const { done, value } = await reader.read()
-
-          if (done) {
-            serverResponse.end()
-            break
-          }
-
-          serverResponse.write(value)
-        }
+        serverResponse.socket?.on('error', () => {})
```

Comment:
There isn't an `error` handler on the mock socket we're using on the `serverResponse`. When we pipe the response body to the server response and there's an error, the error event is propagated to the underlying socket. If we don't handle that, we get an unhandled event error and the Node.js kills the process.